src/librustc_mir/interpret/validation.rs

```diff
@@ -175,7 +175,7 @@ impl<'a, 'tcx, M: Machine<'tcx>> EvalContext<'a, 'tcx, M> {
                 }
             }
 
-            let result = self_.resolve_type_vars_if_possible(result);
+            let result = self_.resolve_type_and_region_vars_if_possible(result);
```

Comment:
So should I be revert the change to `type_and_region_vars` as well? (And then I guess squash the commits to avoid needless reverts; GitHub insists on a merge/rebase anyway.)